### PR TITLE
Optimizes processing machinery

### PR DESCRIPTION
**Pull request by DominikPanic**

### CODE REVIEW

File: code/game/machinery/bots/cleanbot.dm [hunk 1]
```diff
@@ -29,23 +29,21 @@
 	var/next_dest
 	var/next_dest_loc
 
-/obj/machinery/bot/cleanbot/New()
```
Comment (spookydonut) [inline code/game/machinery/bots/cleanbot.dm]: atomise

File: code/game/machinery/bots/floorbot.dm [hunk 2]
```diff
@@ -26,10 +26,10 @@
 	var/targetdirection
 
 
-/obj/machinery/bot/floorbot/New()
-	..()
-	src.updateicon()
```
Comment (spookydonut) [inline code/game/machinery/bots/floorbot.dm]: atomise

File: code/game/machinery/bots/cleanbot.dm (hunk 1 above)
Comment (DominikPanic) [reply code/game/machinery/bots/cleanbot.dm]: Eh it's not that bad of an idea to do this, personally I hate making 3 PRs for one simple, coherent change

File: code/game/machinery/bots/cleanbot.dm [hunk 3]
```diff
@@ -29,23 +29,21 @@
...
-	..()
-	src.get_targets()
-	src.icon_state = "cleanbot[src.on]"
+/obj/machinery/bot/cleanbot/Initialize()
+	. = ..()
+	get_targets()
+	icon_state = "cleanbot[src.on]"
```
Comment (Rohesie) [inline code/game/machinery/bots/cleanbot.dm]:
```suggestion
	icon_state = "cleanbot[on]"
```

File: code/game/machinery/door_display/door_display.dm [hunk 4]
```diff
@@ -21,19 +21,17 @@
...
-/obj/machinery/door_display/New()
-	..()
+/obj/machinery/door_display/Initialize()
+	. = ..()
+
+	for(var/obj/machinery/door/D in GLOB.machines)
+		if (D.id == id)
+			targets += D
+
+	if(targets.len == 0)
```
Comment (Rohesie) [inline code/game/machinery/door_display/door_display.dm]:
```suggestion
	if(!length(targets))
```

File: code/game/machinery/vending/vending.dm [hunk 5]
```diff
@@ -662,8 +664,9 @@
 	if(!src.active)
 		return
 
-	if(src.seconds_electrified > 0)
-		src.seconds_electrified--
+	if(seconds_electrified > 0)
+		if(!--seconds_electrified)
+			stop_processing()
```
Comment (Rohesie) [inline code/game/machinery/vending/vending.dm]: Intended?

**Comment by Rohesie** [comment]
The PR is big, and while atomization would be preferable it's not large enough that it makes it absolutely necessary.
Reviewed, it looks ready after minor changes and testing.

**Comment by DominikPanic** [comment]
It's been tested since it was made xP

**Pull request merged**